### ISSUE TITLE
[Grid - Basics] Column does have empty header

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3181]
## Current behavior

In [Menu Button story](https://storybook.sanomalearning.design/?path=/story/grid-basics--menu-button) third column has header `th role="columnheader" part="header menu-button"` which is empty.
Screen reader users won't get information about what to expect in that column.

<img width="660" height="458" alt="Image" src="https://github.com/user-attachments/assets/3febe289-be03-4f82-a353-e8f4c26d7041" />

# Suggested solution

It's recommended to add text hidden visually with CSS, but announced by screen readers to this header. This is the best solution as it would work the same on different browsers and for different screen readers. This also fixes axe-core violation "empty table header"

Text of header should be "Options" or something similar that explains to user what items are in that column.

Adding `aria-label` to the header is not recommended because it may work for VoiceOver but don't work for NVDA and also axe-core automated tests still find violation with empty header then. 

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
